[IIO] enable buffers

kparichay opened the pull request:
1. Enabling buffers in _start
2. Setting channels data location in buffers
3. Storing only filtered channels
Added some bugfixes and fixes to comments

Signed-off-by: Parichay Kapoor <pk.kapoor@samsung.com>

myungjoo commented on gst/nnstreamer/tensor_source/tensor_src_iio.c:
```diff
@@ -158,6 +159,7 @@ enum
  * @brief IIO system paths
  */
 #define IIO_BASE_DIR "/sys/bus/iio/devices/"
+#define IIO_DATA_DIR "/dev/"
```
This name may misleading readers because "datadir" usually means "/usr/share/...". You may want to use "IIO_DEV_DIR" or "DEV_DIR" instead.

kparichay replied:
Done.

myungjoo commented on gst/nnstreamer/tensor_source/tensor_src_iio.c:
```diff
@@ -870,6 +895,35 @@ gst_tensor_set_all_channels (GstTensorSrcIIO * self, const gint contents)
 
+/**
+ * @brief get the size of the combined data from channels
+ * @param[in] self List of all the channels
```
self --> channels ??

kparichay replied:
done.

kparichay commented:
**V2**:
Updated `IIO_DATA_DIR` -> `IIO_DEV_DIR`
updated doxygen comment for `self`->`channels`

**V3*:
Added disabling of buffers in _stop()

myungjoo commented on gst/nnstreamer/tensor_source/tensor_src_iio.c:
```diff
@@ -402,9 +404,11 @@ gst_tensor_src_iio_set_channel_type (GstTensorSrcIIOChannelProperties * prop,
       sscanf (contents, "%ce:%c%u/%u>>%u", &endianchar, &signchar,
-      &prop->mask_bits, &prop->storage_bits, &prop->shift);
+      &prop->mask_bits, &storage_bits, &prop->shift);
+  prop->storage_bytes = storage_bits >> 3;
```
What if ```(stroage_bits & 0x7 > 0) == TRUE```? Don't we need to round-up?
(e.g., ``` ((stroage_bits - 1) >> 3) + 1 if storage_bits > 0```)?

kparichay replied:
Yes, I agree. 
Also, need to store storage_bits along with storage bytes to handle signed data when `(storage_bits & 0x7 > 0)` .

kparichay commented:
**V2:**
Updated IIO_DATA_DIR -> IIO_DEV_DIR
updated doxygen comment for self->channels

**V3:**
Added disabling of buffers in _stop()

**V4:**
Handled case where `(storage_bits & 0x7 > 0)`
Added storage_bits in ChannelProperties struct to handle signed data, when `(storage_bits & 0x7 > 0)`

myungjoo commented on gst/nnstreamer/tensor_source/tensor_src_iio.c:
```diff
@@ -405,6 +407,10 @@ gst_tensor_src_iio_set_channel_type (GstTensorSrcIIOChannelProperties * prop,
       &prop->mask_bits, &prop->storage_bits, &prop->shift);
+  if (prop->storage_bits > 0) {
+    prop->storage_bytes = ((prop->storage_bits - 1) >> 3) + 1;
+  }
```
```
else
  prop->storage_bytes = 0; /* This may be not initialized */
```

kparichay replied:
added

kparichay commented:
**V5**
- added else case for storage_bytes to 0
- removed bug - freeing memory when filtering enabled channels
- removed bug - using g_list_remove() to remove data than g_list_remove_link()
